src/messageComposer/attachmentManager.ts

```diff
@@ -473,21 +482,7 @@ export class AttachmentManager {
     try {
       response = await this.doUploadRequest(localAttachment.localMetadata.file);
     } catch (error) {
-      let finalError: Error = {
-        message: 'Error uploading attachment',
-        name: 'Error',
-      };
-      if (typeof (error as Error).message === 'string') {
-        finalError = error as Error;
-      } else if (typeof error === 'object') {
-        finalError = Object.assign(finalError, error);
-      }
-
-      this.client.notifications.addError({
-        message: finalError.message,
-        origin: { emitter: 'AttachmentManager', context: { attachment } },
-      });
-
+      const reason = (error as Error).message ?? 'unknown error';
```

Comment:
```suggestion
      const reason = error instanceOf Error ? error.message : 'unknown error';
```

Reply:
Not inherently better, just "type-safer" - either of the options do not handle errors which come back as simple strings though. 

![image](https://github.com/user-attachments/assets/9010495e-456f-4b54-ad5c-58d1a2f0a885)